skyfield/sgp4lib.py

```diff
@@ -153,6 +153,30 @@ def ITRF_position_velocity_error(self, t):
         rITRF, vITRF = TEME_to_ITRF(t.ut1, rTEME, vTEME)
         return rITRF, vITRF, error
 
+    def is_sunlit(self, ephemeris, times):
+        """Calculates if the satellite is currently sunlit (True/False) at each time in times. Returns an array of booleans."""
+        Re = 6378.137  # Equatorial radius of the Earth in km
+        earth = ephemeris['earth']
+        sun = ephemeris['sun']
+        sat = earth + self
+
+        sunpos, earthpos, satpos = [
+            thing.at(times).position.km for thing in (sun, earth, sat)
+        ]
+        sunearth, sunsat = earthpos-sunpos, satpos-sunpos
+        sunearthnorm, sunsatnorm = [
+            vec/sqrt((vec**2).sum(axis=0)) for vec in (sunearth, sunsat)
+        ]
+        angle = arccos((sunearthnorm * sunsatnorm).sum(axis=0))
+        sunearthdistance = sqrt((sunearth**2).sum(axis=0))
+        sunsatdistance = sqrt((sunsat**2).sum(axis=0))
+        limbangle = arctan2(Re, sunearthdistance)
```

Comment:
The diagram I've sketched suggests that `arctan2()` does not work here? I'll try it in ASCII art (gulp!):

```
Sun

|\ <--- angle
| \
|  \
|   \   sunearthdistance
|_   \
|_|___\
       Earth center

 Earth
 radius
```
Feel free to follow up with another diagram if I've imagined the triangle a bit wrong.